File: showdown-go/entity/player/ai_player.go
package player

import (
	"github.com/go-faker/faker/v4"
	"math/rand"
	"showdown-go/entity"
)

type AIPlayer struct {
	AbstractPlayer
}

func NewAIPlayer(showdown *entity.Showdown) *AIPlayer {
	return &AIPlayer{
		AbstractPlayer: NewAbstractPlayer(*showdown),
	}
}

func (a *AIPlayer) ChooseAction() {
	// Randomly choose action
	if rand.Intn(2) == 1 {
		// exchange card and can't exchange myself
		num := rand.Intn(len(a.Showdown.GetPlayers()))
		player := a.Showdown.GetPlayers()[num]

		for player.GetName() == a.GetName() {
			num := rand.Intn(len(a.Showdown.GetPlayers()))
			player = a.Showdown.GetPlayers()[num]
		}

		exchangeHand := entity.NewExchangeHands(a, player)
		a.Showdown.AddExchangeHands(exchangeHand)
		a.SetExchangePermission(false)
	}

	targetPlayer := a.CheckChangeHands()
	targetCard := rand.Intn(len(targetPlayer.GetHand()))
	card := targetPlayer.GetHand()[targetCard]
	a.ShowCard(card)

	// Remove card from hand
	targetPlayer.GetHand()[targetCard] = nil
	targetPlayer.SetHand(targetPlayer.GetHand())
}

func (a *AIPlayer) NameHimself() {
	a.SetName(faker.LastName())
}

File: C2MB/src/Players/AIPlayer.php
<?php

namespace C2MB\Players;

use C2MB\ExchangeHands;
use Faker\Factory;

class AIPlayer extends Player
{
    public function chooseAction(): void
    {
        $players = $this->showdown->getPlayers();

        // If there are more than one player and exchange permission then choose action
        if (count($players) > 1 && $this->exchangePermission) {
            // Randomly choose action
            if (rand(0, 1)) {
                // exchange card and can't exchange myself
                /** @var HumanPlayer|self $player */
                $num = array_rand($players);
                $player = $players[$num];

                while ($player->getName() !== $this->getName()) {
                    $num = array_rand($players);
                    $player = $players[$num];
                }

                $exchangeHand = new ExchangeHands($this, $player);
                $this->showdown->addExchangeHands($exchangeHand);
                $this->setExchangePermission(false);
            }
        }

        $targetPlayer = $this->checkChangeHands();
        $targetCard = array_rand($targetPlayer->getHand());
        $card = $targetPlayer->hand->cards[$targetCard];
        $this->showCard($card);

        // Remove card from hand
        unset($targetPlayer->hand->cards[$targetCard]);
        $targetPlayer->hand->cards = array_values($this->hand->cards);
    }

    public function nameHimself(): void
    {
        $faker = Factory::create();
        $this->setName($faker->lastName);
    }

    public function takeAction(array $actions): int
    {
        return array_rand($actions);
    }

    public function generateName(): string
    {
        $faker = Factory::create();

        return $faker->name;
    }
}
